velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -2858,6 +2858,46 @@ TEST_F(DateTimeFunctionsTest, dateParse) {
       dateParse("116", "%y+"), "Invalid format: \"116\" is malformed at \"6\"");
 }
 
+// The test for the alias now() is integrated into this test.
+TEST_F(DateTimeFunctionsTest, currentTimestamp) {
+  const auto evaluateCurrentTimestamp = [&](const std::string& functionName,
+                                            int64_t expectedTimestamp,
+                                            const std::string& timeZone) {
+    auto emptyRowVector = makeRowVector(ROW({}), 1);
+    auto actual = evaluate(functionName, emptyRowVector);
+    auto expected =
+        makeTimestampWithTimeZoneVector(expectedTimestamp, timeZone.c_str());
+
+    ASSERT_EQ(expected->size(), actual->size());
```

Comment:
In the past we just checked the value at index 0 since the vector had only one element (from sizing at line 2866). Why was that changed ?

Reply:
Since the second element is the timezone, we also want to check the timezone is the same.

Reply:
It might be better to add 2 ASSERT_EQ one for the timestamp and another for the timezone, so that the intent is more clear. Here, its a bit confusing about what is being checked.

Reply:
The reason why I am not using ASSERT_EQ is it can't allow lag for timstamp like `ASSERT_TRUE(actual->compare(expected.get(), i, i) <= allowedLag)`. I can add a comment to tell what is the first and second element in the vector.

Reply:
This part of the code is doing the same thing as `assertEqualVectors()', except it allows lags.

Reply:
We needn't use ASSERT_EQ for timestamp (and use the format you have specified) and then have ASSERT_EQ for timezones.

On account of the iteration, the error messages also give the index, so its a bit confusing.
Since we have only one timestamp returned having one statement checking timestamp and another statement asserting the timezone it simple and precise. The iteration is unnecessary.

Another option for you is to use evaluateOnce instead of evaluate for a cleaner test.

Reply:
I mean below function in /root/velox/velox/vector/tests/utils/VectorTestBase.cpp. It can access the index 1, the code I wrote is the same but it can't access index 1
```
void assertEqualVectors(const VectorPtr& expected, const VectorPtr& actual) {
  ASSERT_EQ(expected->size(), actual->size());
  ASSERT_TRUE(expected->type()->equivalent(*actual->type()))
      << "Expected " << expected->type()->toString() << ", but got "
      << actual->type()->toString();
  for (auto i = 0; i < expected->size(); i++) {
    ASSERT_TRUE(expected->equalValueAt(actual.get(), i, i))
        << "at " << i << ": expected " << expected->toString(i) << ", but got "
        << actual->toString(i);
  }
}
```